packages/teleterm/src/mainProcess/mainProcess.ts

```diff
@@ -203,6 +203,52 @@ export default class MainProcess {
       this.windowsManager.forceFocusWindow();
     });
 
+    // Used in the `tsh install` command on macOS to make the bundled tsh available in PATH.
+    // Returns true if tsh got successfully installed, false if the user closed the osascript
+    // prompt. Throws an error when osascript fails.
+    ipcMain.handle('main-process-symlink-tsh', async () => {
+      const source = this.settings.tshd.binaryPath;
+      const target = '/usr/local/bin/tsh';
+      const prompt =
+        'Teleport Connect wants to create a symlink for tsh in /usr/local/bin.';
+      const command = `osascript -e "do shell script \\"mkdir -p /usr/local/bin && ln -sf '${source}' '${target}'\\" with prompt \\"${prompt}\\" with administrator privileges"`;
+
+      try {
+        await promisify(exec)(command);
+        this.logger.info(`Created the symlink to ${source} under ${target}`);
+        return true;
+      } catch (error) {
+        // Ignore the error if the user canceled the prompt.
+        // https://developer.apple.com/library/archive/documentation/AppleScript/Conceptual/AppleScriptLangGuide/reference/ASLR_error_codes.html#//apple_ref/doc/uid/TP40000983-CH220-SW2
+        if (error instanceof Error && error.message.includes('-128')) {
+          return false;
+        }
+        this.logger.error(error);
+        throw error;
+      }
+    });
+
+    ipcMain.handle('main-process-remove-tsh-symlink', async () => {
+      const target = '/usr/local/bin/tsh';
+      const prompt =
+        'Teleport Connect wants to remove a symlink for tsh from /usr/local/bin.';
+      const command = `osascript -e "do shell script \\"rm '${target}'\\" with prompt \\"${prompt}\\" with administrator privileges"`;
+
+      try {
+        await promisify(exec)(command);
+        this.logger.info(`Removed the symlink under ${target}`);
+        return true;
+      } catch (error) {
+        // Ignore the error if the user canceled the prompt.
+        // https://developer.apple.com/library/archive/documentation/AppleScript/Conceptual/AppleScriptLangGuide/reference/ASLR_error_codes.html#//apple_ref/doc/uid/TP40000983-CH220-SW2
+        if (error instanceof Error && error.message.includes('-128')) {
+          return false;
+        }
+        this.logger.error(error);
+        throw error;
```

Comment:
I have mixed feelings about this, it would be nice to return the errors in the same way.
Maybe we could create a new custom error, like `PromptCancelledByUserError` and throw (or even return) it instead of `false`? Then on the renderer side we would have to create our own `instanceOf` (as the the classes are serialized) that would compare errors by `name`.

IMO this would make the code more elegant, but we can live with what we have, it is quite simple and documented.

Reply:
> I have mixed feelings about this, it would be nice to return the errors in the same way.

The way I think about this is that closing the prompt is not really an error/exception so I wanted to reserve rejecting the promise for actual errors that might happen during the execution of the script.

As seen in your suggestion, using a custom error would be much more complicated. But even more important is that it wouldn't work because errors are not only stripped of any custom properties, they lose their name as well and become simple `Error` instances. So we couldn't inspect `name` AFAIR.

Reply:
So the only way is to return plain objects, but if we don't think of closing the prompt as an error then it's fine to stay with the current solution.